### PR TITLE
Add ForwardHeaderOptions

### PR DESCRIPTION
**Pull request by MichaelRobinson-BJSS**
Change auth redirect to being relative path

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -78,11 +80,21 @@ public static void ConfigureCookies(this IServiceCollection services, IConfigura
...
                 options.ExpireTimeSpan = cookieExpiration.ExpireTimeSpan;
                 options.SlidingExpiration = cookieExpiration.SlidingExpiration;
                 options.AccessDeniedPath = "/404";
+
+                options.Events = new CookieAuthenticationEvents
+                {
+                    OnRedirectToLogin = ctx =>
+                    {
+                        var relativeRedirectUri = new Uri(ctx.RedirectUri).PathAndQuery;
+                        ctx.Response.Headers["Location"] = relativeRedirectUri;
```
Comment (bjss-chris) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/ServiceCollectionExtensions.cs:93]: Recommend using `HeaderNames.Location`.

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -78,11 +80,21 @@ public static void ConfigureCookies(this IServiceCollection services, IConfigura
...
+                        ctx.Response.StatusCode = 302;
```
Comment (bjss-chris) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/ServiceCollectionExtensions.cs:94]: Recommend using `StatusCodes.Status302Found`.

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Startup.cs [hunk 3]
```diff
@@ -100,6 +100,15 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env, IMapper
         {
             app.UseResponseCompression();
 
+            var forwardingOptions = new ForwardedHeadersOptions()
```
Comment (bjss-chris) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Startup.cs:103]: Brackets can be omitted.

**Review by bjss-chris — CHANGES_REQUESTED** [review]
Couple of small changes to improve readability.

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/ServiceCollectionExtensions.cs:93]: Done

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/ServiceCollectionExtensions.cs (hunk 2 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/ServiceCollectionExtensions.cs:94]: Done

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Startup.cs (hunk 3 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Startup.cs:103]: Done

**Review by bjss-chris — APPROVED** [review]

**Pull request merged**